### PR TITLE
AP-3804: Add state benefits loop for partner

### PR DESCRIPTION
**Pull request by colinbruce**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/AP-3804)

This replicates the state_benefit loop from the client pages.  
It refactors the shared views as much as possible to reduce duplication

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing: `bundle exec rake`
- Github should not be reporting conflicts; you should have recently run `git rebase main`.
- There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- The PR description should say what you changed and why, with a link to the JIRA story.
- You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: app/views/shared/state_benefits/_add_other_state_benefits.html.erb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    <span class="govuk-caption-l"><%= caption %></span>
+    <h1 class="govuk-heading-xl"><%= t(".existing", count: pluralize(@transactions.count, "benefit").to_s) %></h1>
+    <%= govuk_summary_list(classes: "summary_list_action_width_auto") do |summary_list| %>
+      <% @transactions.order(:created_at).each_with_index do |transaction, index| %>
+        <%= summary_list.with_row(html_attributes: { id: "transaction_#{index}" }) do |row| %>
+          <%= row.with_key { transaction.description } %>
+          <%= row.with_value { "#{gds_number_to_currency(transaction.amount)} #{t("transaction_types.frequencies.#{transaction.frequency}").downcase}" } %>
+          <%= row.with_action(
+                text: t("generic.change"),
+                href: url_helper.send("providers_legal_aid_application_#{url_part}_state_benefit_path", @legal_aid_application, transaction),
```
Comment (jsugarman) [inline app/views/shared/state_benefits/_add_other_state_benefits.html.erb:17]: i find the explicit passing of the existing/default url helpers odd, but can see why you need it for the `send`.

I wonder if you can just pass url another param like `show_url: providers_legal_aid_application_means_state_benefit_path || providers_legal_aid_application_partners_state_benefit_path` instead
of `url_part` 

And use
```
href: show_url(@legal_aid_application, transaction)
```

**Review by jsugarman — APPROVED** [review]

File: app/views/providers/partners/add_other_state_benefits/show.html.erb [hunk 2]
```diff
@@ -1,36 +1,7 @@
...
-                                              legend: { text: content_for(:page_title), size: "m", tag: "h2" } %>
-
-      <%= next_action_buttons(form:) %>
-  <% end %>
-<% end %>
+<%= render "shared/state_benefits/add_other_state_benefits",
+           form: @form,
+           page_title: t(".page_title"),
+           caption: t("generic.partner_means_caption"),
+           url_part: "partners",
```
[the remark this answers is not in the document]
Comment (colinbruce) [reply app/views/providers/partners/add_other_state_benefits/show.html.erb:5]: Wouldn't that be nice! 😃 
Once the partner stuff is complete we could investigate refactoring more of the means logic into a clearer namespace :+1:

**Review by RoseSAK — APPROVED** [review]
Looks good! Only one minor comment 🎉

File: app/helpers/state_benefit_partial_url_helper.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+module StateBenefitPartialUrlHelper
+  def state_benefit_partial_url(type, version, *)
```
[the remark this answers is not in the document]
Comment (RoseSAK) [reply app/helpers/state_benefit_partial_url_helper.rb:2]: Ahh I see yeah 😬 having just read the above comments (which I didn't notice earlier!) I would agree to just leave it for now and consider refactoring all the naming later on....

**Pull request merged**